src/pg_to_evalscript/javascript_processes/resample_cube_temporal.js

```diff
@@ -0,0 +1,200 @@
+      let minDiff = Number.POSITIVE_INFINITY;
+      let minDifParsedDataLabel = null;
+      let nearestDataLabelIndex = 0;
+      let dataLabelIndicesWithin = [];
+
+      // find the nearest SOURCE label (date) for the current TARGET label (date)
+      for (let dataLabelIndex in dataDim.labels) {
+        dataLabelIndex = parseInt(dataLabelIndex);
+
+        let parsedDataLabel = new Date(parse_rfc3339(dataDim.labels[dataLabelIndex]).value);
+        let diff = Math.abs(parsedDataLabel - parsedTargetLabel);
+
+        if (diff < minDiff) {
+          minDiff = diff;
+          minDifParsedDataLabel = parsedDataLabel;
+          nearestDataLabelIndex = dataLabelIndex;
+        }
+
+        // DOCS: The rare case of ties is resolved by choosing the earlier timestamps.
+        if (diff === minDiff) {
+          if (parsedDataLabel < minDifParsedDataLabel) {
+            minDiff = diff;
+            minDifParsedDataLabel = parsedDataLabel;
+            nearestDataLabelIndex = dataLabelIndex;
+          }
+        }
+
+        // if valid_within is set, save all the labels that are within 
+        // the timespan [targetLabel +/- valid_within]
+        // and sort them by distance from targetLabel afterwards
+        if (valid_within !== undefined && valid_within !== null) {
+          const diffInDays = diff / (1000 * 60 * 60 * 24);
+          if (diffInDays <= valid_within) {
+            dataLabelIndicesWithin.push({ dataLabelIndex, diffInDays })
+          }
+        }
+      }
+
+      // pick the correct values (sub-ndarray) from SOURCE's ndarray
+      const shapeArrayWithNullExceptLabelIndex = dataClone.dimensions.map(
+        (d, i) => i == dataDimIndex ? nearestDataLabelIndex : null
+      );
+      const pickedNdarray = dataClone.data.pick(...shapeArrayWithNullExceptLabelIndex);
+      const flatPickedNdarray = flattenToNativeArray(pickedNdarray, true);
+
+      let dataToAdd = flatPickedNdarray;
+
+      // if valid_within is set, the values to add might be from the labels (dates) that are
+      // outside of the timespan [targetLabel +/- valid_within]
+      if (valid_within !== undefined && valid_within !== null) {
+        if (dataLabelIndicesWithin.length === 0) {
+          dataToAdd = dataToAdd.map(el => null);
+        }
+        else {
+          dataLabelIndicesWithin.sort((a, b) => a.diffInDays - b.diffInDays);
+          for (let dl of dataLabelIndicesWithin) {
+            // pick the correct values (sub-ndarray) from SOURCE's ndarray
+            const shapeArrayWithNullExceptLabelIndex = dataClone.dimensions.map(
+              (d, i) => i == dataDimIndex ? dl.dataLabelIndex : null
+            );
+            const pickedNdarray = dataClone.data.pick(...shapeArrayWithNullExceptLabelIndex);
+            const flatPickedNdarray = flattenToNativeArray(pickedNdarray, true);
+
+            for (idx in dataToAdd) {
+              if (!is_valid({ x: dataToAdd[idx] })) {
```

Comment:
If you mean that unit tests in `test_resample_cube_temporal.py` should fail - I added a function `load_dependency_processes_code` in tests/utils.py to naively load the code for the dependency processes. 

I will remove it and use your suggestion for integration tests.

Reply:
No, it would fail in the generated evalscript as the code for `is_valid` isn't loaded anywhere. There is an explanation here: https://github.com/openEOPlatform/openeo-pg-evalscript-converter/pull/36#pullrequestreview-897287996

Reply:
Hopefully this is now fixed.